community/cypher/acceptance/src/test/scala/org/neo4j/internal/cypher/acceptance/NaNAcceptanceTest.scala

```diff
@@ -0,0 +1,369 @@
+
+    val result = executeWithAllPlanners(query)
+
+    shouldContainOnlyNullFor(result, "shouldBeNull")
+  }
+
+  test("asin() outside of [-1, 1] should not produce nan") {
+    val query = "WITH asin(2) AS shouldBeNull RETURN shouldBeNull"
+
+    val result = executeWithAllPlanners(query)
+
+    shouldContainOnlyNullFor(result, "shouldBeNull")
+  }
+
+  test("acos() outside of [-1, 1] should not produce nan") {
+    val query = "WITH acos(2) AS shouldBeNull RETURN shouldBeNull"
+
+    val result = executeWithAllPlanners(query)
+
+    shouldContainOnlyNullFor(result, "shouldBeNull")
+  }
+
+  test("0 over 0 should not produce nan") {
+    val query = s"WITH 0.0 / 0.0 AS shouldBeNull RETURN shouldBeNull"
+
+    a [org.neo4j.cypher.ArithmeticException] should be thrownBy executeWithAllPlanners(query)
+  }
+
+  test("any other number over 0 should produce a too large number") {
+    val query = s"WITH 10.0 / 0.0 AS inf RETURN inf"
+
+    a [SyntaxException] should be thrownBy executeWithAllPlanners(query)
+  }
+
+  test("square root of negative number should not produce nan with parameter") {
+    val query = "WITH sqrt({param}) AS shouldBeNull RETURN shouldBeNull"
+
+    val result = executeWithAllPlanners(query, "param" -> -1)
+
+    shouldContainOnlyNullFor(result, "shouldBeNull")
+  }
+
+  test("log of negative number should not produce nan with parameter") {
+    val query = "WITH log({param}) AS shouldBeNull RETURN shouldBeNull"
+
+    val result = executeWithAllPlanners(query, "param" -> -1)
+
+    shouldContainOnlyNullFor(result, "shouldBeNull")
+  }
+
+  test("log10 of negative number should not produce nan with parameter") {
+    val query = "WITH log10({param}) AS shouldBeNull RETURN shouldBeNull"
+
+    val result = executeWithAllPlanners(query, "param" -> -1)
+
+    shouldContainOnlyNullFor(result, "shouldBeNull")
+  }
+
+  test("asin() outside of [-1, 1] should not produce nan with parameter") {
+    val query = "WITH asin({param}) AS shouldBeNull RETURN shouldBeNull"
+
+    val result = executeWithAllPlanners(query, "param" -> 2)
+
+    shouldContainOnlyNullFor(result, "shouldBeNull")
+  }
+
+  test("acos() outside of [-1, 1] should not produce nan with parameter") {
+    val query = "WITH acos({param}) AS shouldBeNull RETURN shouldBeNull"
+
+    val result = executeWithAllPlanners(query, "param" -> 2)
+
+    shouldContainOnlyNullFor(result, "shouldBeNull")
+  }
+
+  test("0 over 0 should not produce nan with parameters") {
+    val query = s"WITH {lhs} / {rhs} AS shouldBeNull RETURN shouldBeNull"
+
+    a [org.neo4j.cypher.ArithmeticException] should be thrownBy executeWithAllPlanners(query, "lhs" -> 0.0, "rhs" -> 0.0)
+  }
+
+  test("any other number over 0 should produce a too large number with parameters") {
+    val query = s"WITH {lhs} / {rhs} AS inf RETURN inf"
+
+    a [org.neo4j.cypher.ArithmeticException] should be thrownBy executeWithAllPlanners(query, "lhs" -> 10.0, "rhs" -> 0.0)
```

Comment:
Why should we get a different exception with parameters than with literals? I prefer this exception to the one on line 76.